### ISSUE TITLE
Styles with classes affect controls without such classes

### ISSUE THREAD

**Issue by kekekeks** [AvaloniaUI/Avalonia#4403]
in App.xaml:
```xml
        <Style Selector="ContentPresenter.someclass">
            <Setter Property="Content" Value="{TemplateBinding Content}" />
            <Setter Property="ContentTemplate" Value="{TemplateBinding ContentTemplate}" />
            
        </Style>
```

In MainWindow.xaml:
```xml
<ContentPresenter />
```

```
System.InvalidOperationException: AvaloniaValue.UnsetValue is not a valid value for BindingValue<>.
  at at Avalonia.Data.BindingValue`1.ValidateValue(T value)
  at at Avalonia.Data.BindingValue`1..ctor(T value)
  at at Avalonia.Data.BindingValue`1.op_Implicit(T value)
  at at Avalonia.Styling.PropertySetterBindingInstance`1.ConvertAndPublishNext(Object value)
  at at Avalonia.Styling.PropertySetterBindingInstance`1.Inner.OnNext(Object value)
  at at Avalonia.Reactive.SingleSubscriberObservableBase`1.PublishNext(T value)
  at at Avalonia.Data.TemplateBinding.PublishValue()
  at at Avalonia.Data.TemplateBinding.TemplatedParentChanged()
  at at Avalonia.Data.TemplateBinding.Subscribed()
  at at Avalonia.Reactive.SingleSubscriberObservableBase`1.Subscribe(IObserver`1 observer)
  at at Avalonia.Styling.PropertySetterBindingInstance`1.Subscribed()
  at at Avalonia.Reactive.SingleSubscriberObservableBase`1.Subscribe(IObserver`1 observer)
  at at Avalonia.PropertyStore.BindingEntry`1.Start()
  at at Avalonia.ValueStore.AddBinding[T](StyledPropertyBase`1 property, IObservable`1 source, BindingPriority priority)
  at at Avalonia.AvaloniaObject.Bind[T](StyledPropertyBase`1 property, IObservable`1 source, BindingPriority priority)
  at at Avalonia.Styling.PropertySetterBindingInstance`1.Start(Boolean hasActivator)
  at at Avalonia.Styling.StyleInstance.Start()
  at at Avalonia.Styling.Style.TryAttach(IStyleable target, IStyleHost host)
  at at Avalonia.Styling.Styles.TryAttach(IStyleable target, IStyleHost host)
  at at Avalonia.Styling.Styler.ApplyStyles(IStyleable target, IStyleHost host)
  at at Avalonia.Styling.Styler.ApplyStyles(IStyleable target, IStyleHost host)
  at at Avalonia.Styling.Styler.ApplyStyles(IStyleable target, IStyleHost host)
  at at Avalonia.Styling.Styler.ApplyStyles(IStyleable target)
  at at Avalonia.StyledElement.ApplyStyling()
  at at Avalonia.StyledElement.EndInit()
  at AvaloniaApp.MainWindow.!XamlIlPopulate(IServiceProvider , MainWindow ) in /home/kekekeks/Projects/AvaloniaApp/MainWindow.xaml:10
  at at AvaloniaApp.MainWindow.!XamlIlPopulateTrampoline(MainWindow )
  at AvaloniaApp.MainWindow.InitializeComponent() in /home/kekekeks/Projects/AvaloniaApp/MainWindow.xaml.cs:31
  at AvaloniaApp.MainWindow..ctor() in /home/kekekeks/Projects/AvaloniaApp/MainWindow.xaml.cs:16
  at AvaloniaApp.App.OnFrameworkInitializationCompleted() in /home/kekekeks/Projects/AvaloniaApp/App.xaml.cs:18
  at at Avalonia.Controls.AppBuilderBase`1.Setup()
  at at Avalonia.Controls.AppBuilderBase`1.SetupWithLifetime(IApplicationLifetime lifetime)
  at at Avalonia.ClassicDesktopStyleApplicationLifetimeExtensions.StartWithClassicDesktopLifetime[T](T builder, String[] args, ShutdownMode shutdownMode)
  at AvaloniaApp.Program.Main(String[] args) in /home/kekekeks/Projects/AvaloniaApp/Program.cs:13
```

**Comment by grokys** [comment]
This looks like a bug at the untyped/typed binding value boundary. The reason that the style is being applied that that although the `ContentPresenter` does not currently have `someclass`, it may in future. In this case the style still needs to be applied to preserve style priority.

**Comment by kekekeks** [comment]
It shouldn't crash though

**Comment by grokys** [comment]
Yeah the crash is fixed in #4446.
